ballerina/tests/data_binding_publish_tests.bal

```diff
@@ -617,6 +636,637 @@ service object {
+                if receivedStringValuePublish !is "" {
+                    string receivedMessage = receivedStringValuePublish;
+                    test:assertEquals(receivedMessage, messageToSend, msg = "Message received does not match.");
+                    break;
+                } else {
+                    runtime:sleep(1);
+                    timeoutInSeconds = timeoutInSeconds - 1;
+                }
+            }
+            checkpanic sub.detach(consumerServiceStringPayloadPublish);
+            if timeoutInSeconds == 0 {
+                test:assertFail("Failed to receive the message for 2 minutes.");
+            }
+        } else {
+            test:assertFail("NATS Connection creation failed.");
+        }
+    } else {
+        test:assertFail("NATS Connection creation failed.");
+    }
+}
+
+Service consumerServiceStringPayloadPublish =
+@ServiceConfig {
+    subject: DATA_BINDING_PUBLISH_SUBJECT
+}
+service object {
+    remote function onMessage(string payload) {
+        receivedStringValuePublish = payload;
+        log:printInfo("Message Received: " + receivedStringValuePublish);
+    }
+};
+
+@test:Config {
+    dependsOn: [testDataBindingStringPayloadPublish],
+    groups: ["nats-basic"]
+}
+public function testDataBindingBytesPayloadPublish() {
+    string messageToSend = "Testing Consumer Service With Bytes Data Binding.";
+    Client? newClient = clientObj;
+    if newClient is Client {
+        Listener? sub = listenerObj;
+        if sub is Listener {
+            checkpanic sub.attach(consumerServiceBytesPayloadPublish);
+            checkpanic sub.'start();
+            checkpanic newClient->publishMessage({ content: messageToSend.toBytes(), subject: DATA_BINDING_PUBLISH_SUBJECT });
+            int timeoutInSeconds = 120;
+            // Test fails in 2 minutes if it is failed to receive the message
+            while timeoutInSeconds > 0 {
+                if receivedBytesValuePublish !is "" {
+                    string receivedMessage = receivedBytesValuePublish;
+                    test:assertEquals(receivedMessage, messageToSend, msg = "Message received does not match.");
+                    break;
+                } else {
+                    runtime:sleep(1);
+                    timeoutInSeconds = timeoutInSeconds - 1;
+                }
+            }
+            checkpanic sub.detach(consumerServiceBytesPayloadPublish);
+            if timeoutInSeconds == 0 {
+                test:assertFail("Failed to receive the message for 2 minutes.");
+            }
+        } else {
+            test:assertFail("NATS Connection creation failed.");
+        }
+    } else {
+        test:assertFail("NATS Connection creation failed.");
+    }
+}
 
+Service consumerServiceBytesPayloadPublish =
+@ServiceConfig {
+    subject: DATA_BINDING_PUBLISH_SUBJECT
+}
+service object {
+    remote function onMessage(byte[] payload) {
+        byte[] messageContent = <@untainted> payload;
```

Comment:
```suggestion
        byte[] messageContent = payload;
```

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-nats/pull/427/commits/433577ffabea99c86a5ec34828b469bce521fc85